Week 04/Lecture 08/Assignment 3/crud_employee/src/main/java/com/example/crud_employee/config/DataSourceConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.example.crud_employee.config;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+
+@Configuration
+public class DataSourceConfig {
+
+    @Bean(name = "dataSource1")
+    public DataSource dataSource1() {
+        return DataSourceBuilder.create()
+                .url("jdbc:mysql://localhost:3306/week4_employee")
```

Comment:
All database info should placed in aplication.prop file and use @Value to get value, not hardcode here.

Reply:
I have moved all the database information in `application.properties` and use `@value` annotation in `DataSourceConfig` to get the value.